lms/djangoapps/edraak_misc/tests/test_views.py

```diff
@@ -0,0 +1,41 @@
+"""
+Tests for Edraak Misc.
+"""
+import ddt
+
+from django.core.urlresolvers import reverse
+
+from edraak_tests.tests.helpers import ModuleStoreLoggedInTestCase
+
+
+@ddt.ddt
+class AllTests(ModuleStoreLoggedInTestCase):
+    def setUp(self):
+        super(AllTests, self).setUp()
+
+    @ddt.data('check_student_grades', 'course_complete_status')
+    def test_login_required(self, view):
+        self.client.logout()
+
+        view_full_name = 'edraak_misc:{view}'.format(view=view)
+        url = reverse(view_full_name, args=['edx/Demo/2019'])
+
+        res = self.client.get(url)
+        self.assertRedirects(res, '/login?next={}'.format(url))
+
+    @ddt.data(
+        ('check_student_grades', True, '"success": '),
+        ('course_complete_status', True, '"complete": '),
+        ('all_courses', False, ''),
```

Comment:
Please remove the `all_courses` view, it's deprecated.

```suggestion
```

Reply:
Done